lib/Parse/ParseExpr.cpp

```diff
@@ -1386,6 +1386,55 @@ ExprResult Parser::ParseCastExpression(bool isUnaryExpression,
   // are compiling for OpenCL, we need to return an error as this implies
   // that the address of the function is being taken, which is illegal in CL.
 
+  // With primary-expression, before parsing postfix-expression, check to make
```

Comment:
Could you move your new parsing code into a separate function?   I  think it will make the code easier to maintain and understand in the long run.  I  think it would be good for your comment for the function to provide a more high-level description of what this code did.